### ISSUE TITLE
Enhancement: Report Unicode characters in YARD type specifications instead of crashing

### ISSUE THREAD

**Issue by nijikon** [mensfeld/yard-lint#39]
## Summary

yard-lint should detect and report Unicode characters in YARD type specifications as a new offense type, rather than crashing with an encoding error.

## Current Behavior

When a Ruby file contains Unicode characters in YARD type specifications, such as:

```ruby
# @param [Symbol, …] flags (see {FlagsMap})
def ffi_lib_flags(*flags)
```

yard-lint crashes with an encoding error:

```
invalid byte sequence in UTF-8 (Encoding::CompatibilityError)
```

This happens because YARD's `TypesExplainer::Parser` produces an error message with truncated UTF-8 byte sequences when it encounters non-ASCII characters.

## Expected Behavior

yard-lint should:

1. Detect Unicode/non-ASCII characters in type specifications
2. Report a new offense (e.g., `NonAsciiTypeCharacter` or `UnicodeInType`)
3. Continue processing the rest of the file without crashing

Example output:
```
[W] /path/to/file.rb:137
    NonAsciiTypeCharacter: Type specification contains non-ASCII character '…' (U+2026).
    YARD type specifications should only contain ASCII characters.
    Consider using standard YARD syntax like `Array<Symbol>` for variadic parameters.
```

## Rationale

1. **Linters should report problems, not crash on them** - A malformed type specification is exactly the kind of issue yard-lint should catch
2. **Unicode in types is almost always unintentional** - Usually from copy-paste or attempting non-standard documentation patterns
3. **Provides actionable feedback** - Users can fix the issue instead of debugging encoding crashes
4. **Improves robustness** - yard-lint can process entire codebases (including vendored gems) without failing on edge cases

## Implementation Notes

The detection could happen in the `Tags/TypeSyntax` validator before calling YARD's parser:

1. Check if the type string contains non-ASCII characters
2. If so, report the offense with the specific character and its Unicode code point
3. Skip the YARD parser call for that type (since it will fail anyway)

Alternatively, catch the `SyntaxError` from YARD's parser, detect if it's caused by a Unicode character, and report appropriately.

## Related

This issue was discovered when running yard-lint on a codebase that included vendored gems (specifically `ffi` gem's `library.rb` which uses `…` in documentation).
